### ISSUE TITLE
Build crate documentation

### ISSUE THREAD

**Issue by emilazy** [kolloch/crate2nix#115]
`cargo doc` doesn't behave nicely in the Nix sandbox, since it's basically a kind of build; it'd be nice to have support for something similar in crate2nix (including building docs for all relevant crates).

(I guess this is something `buildRustCrate` would have to handle, but maybe it already does? I couldn't find anything relevant in its docs.)

**Comment by kolloch** [comment]
It would be nice to build the docs in a separate derivation so that it is done in parallel.
